x/incentives/keeper/distribute_test.go

```diff
@@ -1981,10 +1981,11 @@ func (s *KeeperTestSuite) TestAllocateAcrossGauges() {
 }
 
 // This test validates two things:
-// - Allocating to a newly created group without any volume does not panic
-// - Once volume exists, the group allocated correctly.
+// - Allocating to a newly created group does not panic
+// - If groups is modified with invalid values (for testing its refetching), the failure does not occur
+// and the updated group is refetched by AllocateAcrossGauges.
 //
-// Additionally, this test catches a bug where if we didn't refetch groups after
+// This test catches a bug where if we didn't refetch groups after
 // synching in AllocateAcrossGauges, the group would not be updated with the correct
 // total volume and panic when allocating due to division by zero.
 func (s *KeeperTestSuite) TestCreateGroupsAndAllocate_GroupRefetchingInAllocate() {
```

Comment:
Note: I had to slightly refactor this test because it is now impossible to create/allocate to a group with no previous weights set.

I made the goal of the test twofold:
- Test that can allocate to a newly created gauge
- Continue testing that the group is refetched by mutating groups total weight before final allocation